src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -171,7 +189,7 @@ def run_dace_iterator(program: itir.FencilDefinition, *args, **kwargs) -> None:
         sdfg = sdfg_program.sdfg
     else:
         # visit ITIR and generate SDFG
-        program = preprocess_program(program, offset_provider)
+        program = preprocess_program(program, offset_provider, lift_mode)
```

Comment:
This looks good! One note is that regarding DaCe I would probably always create a temporary for all lifts. Since eliminating temporaries is one of the key optimizations, I would find it interesting to see how different approaches deal with it, and if you don't give the chance to DaCe to do this optimization we'll never know.

Reply:
I agree, once temporaries work we can make it the default option.